File: c++/other work/1407B.cpp
#include<iostream>
#include<algorithm>
using namespace std;


int main(){
    int tc;
    cin >>tc;
    for(int j=0;j<tc;j++){
    int n;
    cin>>n;
    int a[n];int max=0,d=0,s=0;
    for(int i=0;i<n;i++){cin>>a[i];if (max<a[i]){max=a[i];s=i;}}
    a[s]=-1;
    sort(a,a+n);
    cout<<max<<" ";
    int m=0;
    for (int k=0;k<n;k++){int gcd=__gcd(max,a[k]);if (m<=gcd){ m=gcd;s=k;}}
    cout<<a[s]
    a[s]=-1;
    for( int i)

    }
}

File: python/Other work/1407B.py
import math
for _ in range(int(input())):
    n=int(input())
    a=sorted(list(map(int,input().split())),reverse=True)
    b=a.pop(0)
    print(b,end=" ")
    if n==1:
        print()
        continue
    max=math.gcd(b,a[0])
    s=0
    for i in range(n-2,-1,-1):
        t=math.gcd(b,a[i])
        if max<t:
            max=t
            s=i
            
    print(a.pop(s),end=" ")
    for i in range(n-2):
        if i%max==0:
            print(a[i],end=" ")
            a[i]=-1
    for i in a:
        if i==-1:
            continue
        print(i,end=" ")
    print()
        
            
        
        
    